master-docker-nonstandard-2/master.cfg

```diff
@@ -152,21 +163,47 @@ f_asan_ubsan_build.addStep(
 # build steps
 f_asan_ubsan_build.addStep(
     steps.ShellCommand(
-        command='echo "leak:libtasn1\nleak:libgnutls\nleak:libgmp" > mysql-test/lsan.supp',
+        name="fetch latest UBSAN/ASAN filters",
+        command="""
+           for san in UBSAN ASAN ; do \\
+             curl https://raw.githubusercontent.com/mariadb-corporation/mariadb-qa/refs/heads/master/"$san".filter >> mysql-test/"${san,,*}".supp; \\
+           done
+         """,
         doStepIf=filterBranch,
     )
 )
+f_asan_ubsan_build.addStep(
+    steps.ShellCommand(command="cat mysql-test/asan.supp", doStepIf=filterBranch)
+)
+f_asan_ubsan_build.addStep(
+    steps.ShellCommand(command="cat mysql-test/ubsan.supp", doStepIf=filterBranch)
+)
 f_asan_ubsan_build.addStep(
     steps.ShellCommand(command="cat mysql-test/lsan.supp", doStepIf=filterBranch)
 )
 f_asan_ubsan_build.addStep(
     steps.Compile(
+        name="build",
         command=[
             "sh",
             "-c",
             util.Interpolate(
-                "cmake . -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=Debug -DWITH_ASAN=YES -DWITH_UBSAN=YES -DPLUGIN_TOKUDB=NO -DPLUGIN_MROONGA=NO -DPLUGIN_OQGRAPH=NO -DPLUGIN_ROCKSDB=NO -DPLUGIN_CONNECT=NO -DWITH_SAFEMALLOC=OFF -DWITH_ZLIB=bundled -DWITH_SSL=bundled -DWITH_DBUG_TRACE=OFF -DWITH_SAFEMALLOC=OFF && make VERBOSE=1 -j%(kw:jobs)s package",
+                """
+                cmake . \\
+                        -DWITH_ASAN=ON \\
+                        -DWITH_ASAN_SCOPED=ON \\
+                        -DWITH_UBSAN=ON \\
+                        -DCMAKE_BUILD_TYPE=%(kw:build_type)s \\
+                        -DCMAKE_C_COMPILER=%(kw:c_compiler)s \\
+                        -DCMAKE_CXX_COMPILER=%(kw:cxx_compiler)s \\
+                        -DWITH_EMBEDDED_SERVER=OFF \\
+                        -DPLUGIN_PERFSCHEMA=NO \\
```

Comment:
note to self, change PERFSCHEMA disable to WITH_UNIT_TESTS=OFF. The perfschema (non-unit) tests are probably more valuable the unit tests.